client/src/app/slides/base-slide-component.ts

```diff
@@ -13,7 +13,9 @@ export abstract class BaseSlideComponent<T extends object> {
      * Each slide must take slide data.
      */
     @Input()
-    public set data(value: SlideData<T>) {}
+    public set data(value: SlideData<T>) {
+        this._data = value;
+    }
 
     public get data(): SlideData<T> {
         return this._data;
```

Comment:
Could this be unified to `@Input() public data: SlideData<T>`?

Reply:
Yes, right. It seems to be a leftover from debugging.

I'll fix this when implementing all other slides.

Reply:
nevermind, fixed.

Reply:
ah, no. It is currently needed by some child components.